File: Python/Basic Syntax Practice/calculator.py
print("===== CALCULATOR =====")
print()

a = float(input("enter first number = "))
b = float(input("enter second number = "))
op = input("enter basic mathematical operation (+,-,*,/,%): ")
print()

result = None

if op == "+" :
    result = a + b
    
elif op == "-" :
    result = a - b
    
elif op == "*" :
    result = a * b
    
elif op == "/" :
    
    if b == 0 :
        print("Invalid input for b")
    else:
        result = a / b
        
elif op == "%" :
    result = a % b

else :
    print("invalid operation input")

if result is not None :
    print(f"result = {result : .4f}")
else : 
    print("error")    

print()    

File: C/Switch/calculator.c
/*input 2 int
 * input math operation (+,-,*,/,%)
 * switch for choosing operation
 * if statement under division by 0
 * print output
 * */

#include <stdio.h>
int main() {
  int x, y;
  float res;
  char op;
  
  printf("enter x,y = ");
  scanf("%d%d", &x, &y);
  printf("enter math operation (+,-,*,/,%) = ");
  scanf(" %c", &op);
  
  switch(op) {
  case '+' : res=x+y;
             printf("result = %.2f\n", res);
             break;
  case '-' : res=x-y;
             printf("result = %.2f\n", res);
             break;
  case '*' : res=x*y;
             printf("result = %.2f\n", res);
             break;
  case '/' : if(y!=0) {
                res=(float)x /  y;
                printf("result = %.2f\n", res);
             }
             else if(y==0) {
                printf("math error\n");
             }
              break;
  case '%' : if(y!=0) {
                int mod_res= x%y;
                printf("result = %d\n", mod_res);
             }
             else if(y==0) {
                printf("math error\n");
             }
              break;
  default :
              printf("invalid input for math operation\n");
              break;
  
  }
  
  return 0;
}